### PR TITLE
Automated separation of images by emotion 

### PR DESCRIPTION
**Pull request by meriam04**
Now in `data_processing.py` there is a `separate_images` function to move images to the correct output directory based on the emotion in the source directory name.

### CODE REVIEW

File: data_processing/data_processing.py [hunk 1]
```diff
@@ -23,3 +29,186 @@ def data_processing(video_path: Path, output_path: Path) -> List[Path]:
 
 if __name__ == "__main__":
     data_processing(Path(sys.argv[1]), Path(sys.argv[2]))
+
+
+def separate_images(source_dirs, output_dir, binary=False):
```
Comment (phonycrow) [inline data_processing/data_processing.py:34]: I like how you combined the binary and multiclass into one function with a boolean `binary` parameter.

File: data_processing/data_processing.py [hunk 2]
```diff
@@ -23,3 +29,186 @@ def data_processing(video_path: Path, output_path: Path) -> List[Path]:
```
Comment (phonycrow) [inline data_processing/data_processing.py:29]: Separate images should be called after `crop_and_resize_images()` here. The function should also take in `binary` as a boolean parameter which could be an additional parameter from the command line.

File: data_processing/data_processing.py [hunk 3]
```diff
@@ -23,3 +29,186 @@ def data_processing(video_path: Path, output_path: Path) -> List[Path]:
...
+        for filename in files:
+            source_file_path = os.path.join(crop_dir, filename)
+            destination_file_path = os.path.join(destination_path, filename)
+            shutil.copy(source_file_path, destination_file_path)
+            logging.debug("Moved %s to %s", filename, destination_path)
+
+    return destination_paths.values()
+
+
+# def separate_images_binary(source_dirs, output_dir):
```
Comment (phonycrow) [inline data_processing/data_processing.py:102]: I like the way it is currently being done. These commented-out versions can be deleted.

File: data_processing/face/tests/test_crop_and_resize_images.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 import pytest
 
 from face.crop_and_resize_images import crop_and_resize_image, crop_and_resize_images
-from utils import Point, Region, Resolution
+from data_processing.utils import Point, Region, Resolution
```
Comment (phonycrow) [inline data_processing/face/tests/test_crop_and_resize_images.py:5]: I think just `from utils import Point, Region, Resolution` should work.

File: data_processing/face/tests/test_separate_images.py [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    logging.debug(os.listdir(setup_folders[0]))
+
+    with pytest.raises(FileNotFoundError) as e:
+        separate_images(source_dirs=setup_folders, output_dir=output_folders)
+
+        assert e.type == error_type
+        assert str(e.value) == error_msg
+
+    if TEAR_DOWN:
+        teardown_test_folders(setup_folders, output_folders)
```
Comment (phonycrow) [inline data_processing/face/tests/test_separate_images.py:264]: The tests look good to me.

**Review by phonycrow — CHANGES_REQUESTED** [review]
PR mostly looks good. There are just a few small changes I recommended.

File: data_processing/face/tests/test_crop_and_resize_images.py (hunk 4 above)
Comment (meriam04) [reply data_processing/face/tests/test_crop_and_resize_images.py:5]: Yeah, that did work. Not sure why I had troubles with it earlier. Thanks!

File: data_processing/data_processing.py (hunk 1 above)
Comment (meriam04) [reply data_processing/data_processing.py:34]: Thanks!

File: data_processing/data_processing.py [hunk 6]
```diff
@@ -2,19 +2,25 @@
...
 from face.crop_and_resize_images import crop_and_resize_images
 from utils import Point, Region, Resolution
 from face.video_to_images import extract_frames
 
+# If we are making a UI, return these values for TOP_LEFT and BOTTOM_RIGHT
 RATE = 1
 TOP_LEFT = Point(430, 80)
 BOTTOM_RIGHT = Point(930, 580)
 RESOLUTION = Resolution(224, 224)
+# Add FILE_LIST as global variable
```
[the remark this answers is not in the document]
Comment (meriam04) [reply data_processing/data_processing.py:19]: You're right, I removed them. Thanks

File: .gitignore [hunk 7]
```diff
@@ -2,6 +2,7 @@ venv/
 data/
 data_processing/face/tests/test_files/cropped
 data_processing/face/tests/test_files/images
+data_processing/face/tests/test_files/separate_images
```
[the remark this answers is not in the document]
Comment (meriam04) [reply .gitignore:5]: Definitely much better. Good suggestion!

File: data_processing/data_processing.py (hunk 2 above)
Comment (meriam04) [reply data_processing/data_processing.py:29]: I added the code to do so. However, when I run `data_processing()` I get an error from the `crop_and_resize_images()` part. I have added the code, but can't test the behaviour at the moment.

**Review by phonycrow — APPROVED** [review]
Thank you for integrating my recommendations!

**Pull request merged**